### PR TITLE
PXC-3589 LOCK=NONE usage is not supported in PXC-5.7 and above (8.0)

### PR DESCRIPTION
**Pull request by patrickbirch**
Merged 5.7 version
Fixed conflicts in source/limitations

### CODE REVIEW

File: doc/source/limitation.rst [hunk 1]
```diff
@@ -82,3 +82,13 @@ Avoid reusing the names of persistent tables for temporary tables
 		<https://dev.mysql.com/doc/refman/8.0/en/temporary-table-problems.html>`_
 
 .. include:: .res/replace.opt.txt
+
+As of version 8.0.25, an INPLACE `ALTER TABLE <https://dev.mysql.com/doc/refman/5.7/en/alter-table.html>`__  query takes an internal shared lock on the table during the execution of the query. The ``LOCK=NONE`` clause is no longer allowed for all of the INPLACE ALTER TABLE queries due to this change.
```
Comment (venkatesh-prasad-v) [inline doc/source/limitation.rst]: We have this limitation 8.0.21. So, It is better we mention 8.0.21 instead of 8.0.25.

Comment (venkatesh-prasad-v) [reply doc/source/limitation.rst]: The link is pointing to 5.7 Docs. Shouldn't we change it to 8.0 Docs?

**Review by venkatesh-prasad-v — CHANGES_REQUESTED** [review]

File: doc/source/limitation.rst (hunk 1 above)
Comment (patrickbirch) [reply doc/source/limitation.rst]: Fixed both.

**Review by venkatesh-prasad-v — APPROVED** [review]
Thanks for addressing the review comments. Approved!

**Pull request merged**